data/datasets/oa_leet10k/oa_leet10k.ipynb

```diff
@@ -0,0 +1,142 @@
+    "]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import kaggle\n",
+    "import pandas as pd\n",
+    "import random\n",
+    "from IPython.display import display\n",
+    "from datasets import Dataset\n",
+    "\n",
+    "data_source = \"https://www.kaggle.com/datasets/erichartford/leetcode-solutions\"\n",
+    "output_dir = \"data\"\n",
+    "os.makedirs(output_dir, exist_ok=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "kaggle.api.dataset_download_files(\"erichartford/leetcode-solutions\", \"data\", unzip=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "leetcode_solutions = pd.read_json(\"data/leetcode-solutions.jsonl\", lines=True)\n",
+    "\n",
+    "# Create dataframe with columns INSTRUCTION, RESPONSE, SOURCE\n",
+    "# The INSTRUCTION a random choice from ONE_STEP_TEMPLATES with the language and content filled in\n",
+    "# The RESPONSE is the answer to the question being posed\n",
+    "# The SOURCE is the URL of the dataset\n",
+    "oa_leet10k = []\n",
+    "for index, row in leetcode_solutions.iterrows():\n",
+    "    content = row[\"content\"]\n",
+    "    for lang in [\"c++\", \"java\", \"javascript\", \"python\"]:\n",
+    "        if lang in row[\"answer\"]:\n",
+    "            oa_leet10k.append(\n",
+    "                {\n",
+    "                    \"INSTRUCTION\": random.choice(ONE_STEP_TEMPLATES)\n",
+    "                    .replace(\"${lang}\", lang)\n",
+    "                    .replace(\"${content}\", content),\n",
+    "                    \"RESPONSE\": row[\"answer\"][lang],\n",
+    "                    \"SOURCE\": data_source,\n",
+    "                }\n",
+    "            )\n",
+    "oa_leet10k = pd.DataFrame(oa_leet10k)\n",
+    "\n",
+    "# Print the first 5 rows of the dataframe with full width and newline characters correctly displayed in the RESPONSE column\n",
+    "with pd.option_context(\"display.max_colwidth\", 80):\n",
+    "    # Assuming the variable df contains the relevant DataFrame\n",
+    "    display(\n",
+    "        oa_leet10k.head(5).style.set_properties(\n",
+    "            **{\n",
+    "                \"text-align\": \"left\",\n",
+    "                \"white-space\": \"pre-wrap\",\n",
+    "            }\n",
+    "        )\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Upload dataset to HF\n",
+    "oa_leet10k.to_parquet(\"oa_leet10k.parquet\", row_group_size=100, engine=\"pyarrow\")\n",
+    "ds = Dataset.from_parquet(\"oa_leet10k.parquet\")\n",
+    "# Uncomment to push dataset to HF\n",
+    "# ds.push_to_hub(\"dctanner/oa_recipes\")"
```

Comment:
Minor point but we could change this to something like `{your_username}/oa_leet10k` to avoid potential confusion/mixups here

Reply:
done